services/venice-controller/src/test/java/com/linkedin/venice/controller/TestDisabledPartitionEnablerService.java

```diff
@@ -30,11 +34,32 @@ public void testCleanupBackupVersion() throws Exception {
     doReturn(true).when(controllerConfig).isEnableDisabledReplicaEnabled();
     doReturn(clusters).when(config).getClusters();
     TestMockTime time = new TestMockTime();
-    DisabledPartitionEnablerService service = new DisabledPartitionEnablerService(admin, config, time);
+    DisabledPartitionEnablerService service = new DisabledPartitionEnablerServiceForTesting(admin, config, time);
 
     service.startInner();
+    latch.await(1, TimeUnit.SECONDS);
     verify(admin, atLeast(1)).enableDisabledPartition(clusterName, "", true);
+    service.stopInner();
+
+  }
+
+  class DisabledPartitionEnablerServiceForTesting extends DisabledPartitionEnablerService {
+    public DisabledPartitionEnablerServiceForTesting(
+        VeniceHelixAdmin admin,
+        VeniceControllerMultiClusterConfig multiClusterConfig,
+        Time time) {
+      super(admin, multiClusterConfig, time);
+      this.cleanupThread = new Thread(new DisabledPartitionEnablerTaskForTesting(), "TestTask");
```

Comment:
This probably isn't very elegant but other ways I've tried would be disruptive. Plz lmk if we have better way to test such cases.